### PR TITLE
feat: conditionally triage issue based on issue labels

### PR DESCRIPTION
**Pull request by peterwoodworth**
At the request of a CDK SDE

Also added linter, and moved output from /lib to /dist

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -14,6 +14,12 @@ inputs:
   default-area:
     description: "Labels and assignees to apply when no keywords are detected"
     required: false
+  included-labels:
+    description: "Only issues which include one of these labels will be triaged"
```
Comment (kellertk) [inline action.yml:18]: Maybe document what happens if this isn't set?

File: src/github.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as github from "@actions/github";
-import * as core from '@actions/core'
-import { IParameter } from './issue'
+import * as core from '@actions/core';
+import { IIssueData } from "./issue";
```
Comment (kellertk) [inline src/github.ts:3]: This seems like we haven't set up a style linter - we're mixing single and double-quote imports. Maybe set up and use something like `google/gts`?

Actually a lot of changes in this file are style stuff that would be caught with a linter.

File: src/github.ts [hunk 3]
```diff
@@ -26,32 +26,41 @@ export class GithubApi {
...
     });
-  
-    content.push(data.title, data.body);
-    return content;
+
+    let title: string = data.title;
+    let body: string = data.body;
+    let labels: string[] = [];
+
+    for(let label of data.labels) {
```
Comment (kellertk) [inline src/github.ts:56]: A linter would also catch things like using `let` instead of `const` here.

File: src/index.ts [hunk 4]
```diff
@@ -8,19 +8,30 @@ async function run() {
...
+  const includedLabels: string[] | undefined = core.getInput('included-labels', { required: false }).replace(/\[|\]/gi, '').split('|');
+  const excludedLabels: string[] | undefined = core.getInput('excluded-labels', { required: false }).replace(/\[|\]/gi, '').split('|');
+
   const issue: Issue = new Issue(content);
-  const winningAreaData: IParameter = issue.getWinningAreaData(issue.determineArea())
+  const winningAreaData: IParameter = issue.getWinningAreaData(issue.determineArea());
+
+  if (includedLabels[0] || excludedLabels[0]) {
+    if (!issue.verifyIssueLabels(includedLabels, excludedLabels)) {
+      console.log("Issue failed label validation. Exiting successfully");
```
Comment (kellertk) [inline src/index.ts:20]: The correct way to log in a github action is to use the `core.log` functions and the like.

**Review by kellertk — CHANGES_REQUESTED** [review]
In addition to my inline comments, I recommend we add the `--license` flag to the build script. This will include a license file in the output folder that contains the licenses of all of the submodules in it. Also we should be publishing to a `dist` folder, not `lib`.

File: src/github.ts (hunk 2 above)
Comment (peterwoodworth) [reply src/github.ts]: linter implemented 👍🏻

File: src/index.ts (hunk 4 above)
Comment (peterwoodworth) [reply src/index.ts]: done!

File: package.json [hunk 5]
```diff
@@ -2,10 +2,15 @@
...
   "scripts": {
-    "build": "ncc build src/index.ts -o lib",
-    "test": "jest"
+    "build": "ncc build src/index.ts -o dist --license LICENSE.md",
+    "test": "jest",
+    "lint": "gts lint",
+    "clean": "gts clean",
+    "compile": "tsc",
+    "fix": "gts fix",
+    "posttest": "npm run lint"
```
Comment (ryparker) [inline package.json:13]: Is `posttest` something that is triggered by GitHub Actions?

**Review by ryparker — APPROVED** [review]
Lots of changes! Looks good to me.

File: package.json (hunk 5 above)
Comment (peterwoodworth) [reply package.json:13]: @ryparker `posttest` runs after `npm run test`

Comment (ryparker) [reply package.json:13]: Ah I didn't know about that hook. Interesting! 👍🏻

**Pull request merged**